mods/ra/maps/fall-of-greece-1-personal-war/personal-war.lua

```diff
@@ -0,0 +1,521 @@
+			a.Move(a.Location + CVec.New(-1,-1))
+			a.Panic()
+		end)
+	end
+	ChurchDamaged = true
+end
+
+SendMig = function(waypoints)
+	local MigEntryPath = { waypoints[1].Location, waypoints[2].Location }
+	local Mig1 = Reinforcements.Reinforce(ussr, { "mig" }, MigEntryPath)
+	if not BridgeBarrel1.IsDead then
+		Utils.Do(Mig1, function(mig)
+			mig.Attack(BridgeBarrel1)
+		end)
+	end
+
+	Trigger.AfterDelay(DateTime.Seconds(5), function()
+		local Mig2 = Reinforcements.Reinforce(ussr, { "mig" }, MigEntryPath)
+		local Mig3 = Reinforcements.Reinforce(ussr, { "mig" }, MigEntryPath)
+		if not CivBarrel.IsDead then
+			Utils.Do(Mig2, function(mig)
+				mig.Attack(CivBarrel)
+			end)
+		end
+		if not DoomedHeli.IsDead then
+			Utils.Do(Mig3, function(mig)
+				mig.Attack(DoomedHeli)
+			end)
+		end
+	end)
+end
+
+SovBaseAttack = function()
+	if not BaseDamaged then
+		local drop3 = RifleDropS.TargetParatroopers(SovBaseDrop.CenterPosition, Angle.East)
+		Utils.Do(drop3, function(a)
+			Trigger.OnPassengerExited(a, function(t, p)
+				IdleHunt(p)
+			end)
+		end)
+
+		Utils.Do(SovBaseTeam, function(actor)
+			if not actor.IsDead then
+				IdleHunt(actor)
+			end
+		end)
+
+		if Map.LobbyOption("difficulty") == "hard" then
+			local barracksTeam = Reinforcements.Reinforce(ussr, RaxTeam, { SovRaxSpawn.Location, SovBaseCam.Location }, 0)
+			Utils.Do(barracksTeam, IdleHunt)
+		end
+	end
+	BaseDamaged = true
+end
+
+ExtractUnits = function(extractionUnit, pos, after)
+	if extractionUnit.IsDead or not extractionUnit.HasPassengers then
+		return
+	end
+
+	extractionUnit.Move(pos)
+	extractionUnit.Destroy()
+
+	Trigger.OnRemovedFromWorld(extractionUnit, after)
+end
+
+SendExtractionHelicopter = function()
+	ExtractionHeli = Reinforcements.ReinforceWithTransport(allies, ExtractionHelicopterType, nil, ExtractionPath)[1]
+	local exitPos = CPos.New(ExtractionPath[1].X, ExtractionPath[2].Y)
+
+	Trigger.OnKilled(ExtractionHeli, function() ussr.MarkCompletedObjective(ussrObj) end)
+	Trigger.OnAllRemovedFromWorld(VIPs, function()
+		ExtractUnits(ExtractionHeli, exitPos, function()
+			allies.MarkCompletedObjective(ProtectVIPs)
+			allies.MarkCompletedObjective(ExtractStavros)
+		end)
+	end)
+end
+
+FindVIPs = function()
+	VIPs = Utils.Where(allies.GetActors(), function(actor)
+		return
+			actor.Type == "e7" or
+			actor.Type == "gnrl" or
+			actor.Type == "e7.noautotarget" or
+			actor.Type == "gnrl.noautotarget"
+	end)
+
+	Trigger.OnAnyKilled(VIPs, function()
+		allies.MarkFailedObjective(ProtectVIPs)
+	end)
+end
+
+WorldLoaded = function()
+	allies = Player.GetPlayer("Allies")
+	ussr = Player.GetPlayer("USSR")
+	badguy = Player.GetPlayer("BadGuy")
+	greece = Player.GetPlayer("Greece")
+	civilians = Player.GetPlayer("GreekCivilians")
+
+	Trigger.OnObjectiveAdded(allies, function(p, id)
+		Media.DisplayMessage(p.GetObjectiveDescription(id), "New " .. string.lower(p.GetObjectiveType(id)) .. " objective")
+	end)
+
+	ussrObj = ussr.AddPrimaryObjective("Kill Stavros.")
```

Comment:
This should use the new `AddObjective` api. While we're at it, it would be good if we could have the objectives and player names start with a capital letter like all other global variables. (It wouldn't be a blocker though if you didn't want to change the player names now, but that is something to keep in mind for when porting the other missions.)